Lib/test/test_asyncio/test_streams.py

```diff
@@ -1129,6 +1129,29 @@ async def inner(httpd):
 
         self.assertEqual(messages, [])
 
+    def test_unclosed_server_resource_warnings(self):
+        async def inner(rd, wr):
+            with self.assertWarns(ResourceWarning) as cm:
+                del wr
+                gc.collect()
+                self.assertEqual(len(cm.warnings), 1)
+                self.assertTrue(str(cm.warnings[0].message).startswith("unclosed <StreamWriter"))
+
+        async def outer():
+            srv = await asyncio.start_server(inner, socket_helper.HOSTv4, 0)
+            async with srv:
+                addr = srv.sockets[0].getsockname()
+                with socket.create_connection(addr):
+                    # Give the loop some time to notice the connection
+                    await asyncio.sleep(0.1)
```

Comment:
Like in the other PR, I believe that one or two sleep(0) should deterministically reach the state you're waiting for?

Reply:
Three, actually.

But I'm cautious about this change. Right now, it seems to require three. But someone might change something so that tomorrow this requires four. At which point this test will start reporting "ok" even if everything is broken. :/

Reply:
I think adding a future to wait on could give something more reliable in this specific case.